src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +680,1187 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+                        {
+                          "scope_string": "https://auth.globus.org/scopes/3f33d83f-ec0a-4190-887d-0622e7c4ee9a/manage",
+                          "allows_refresh_token": False,
+                          "id": "70147193-f88a-4da9-9d6e-677c15e790e5",
+                          "advertised": False,
+                          "required_domains": [],
+                          "name": "Client manage scope",
+                          "description": "Manage configuration of this client",
+                          "client": "3f33d83f-ec0a-4190-887d-0622e7c4ee9a",
+                          "dependent_scopes": [],
+                        },
+                        {
+                          "scope_string": "https://auth.globus.org/scopes/dfc9a6d3-3373-4a6d-b0a1-b7026d1559d6/view",
+                          "allows_refresh_token": False,
+                          "id": "3793042a-203c-4e86-8dfe-17d407d0bb5f",
+                          "advertised": False,
+                          "required_domains": [],
+                          "name": "Client view scope",
+                          "description": "View configuration of this client",
+                          "client": "dfc9a6d3-3373-4a6d-b0a1-b7026d1559d6",
+                          "dependent_scopes": [],
+                        }
+                      ]
+                    }
+
+                    >>> # by all scope ids
+                    >>> scope_ids = ...
+                    >>> r = ac.get_scopes(ids=scopes_ides)
+
+                    >>> # by all scope strings
+                    >>> scope_strings = ...
+                    >>> r = ac.get_scopes(scope_strings=scope_strings)
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.get_scopes
+
+            .. tab-item:: API Info
+
+                ``GET /v2/api/scopes``
+                ``GET /v2/api/scopes?ids=...``
+                ``GET /v2/api/scopes?scope_strings=...``
+
+                .. extdoclink:: Get Scopes
+                    :ref: auth/reference/#get_scopes
+        """  # noqa: E501
+        if scope_strings is not None and ids is not None:
+            raise exc.GlobusSDKUsageError(
+                "AuthClient.get_scopes does not take both "
+                "'scopes_strings' and 'ids'. These are mutually exclusive."
+            )
+
+        if query_params is None:
+            query_params = {}
+
+        if scope_strings is not None:
+            query_params["scope_strings"] = utils.commajoin(scope_strings)
+        if ids is not None:
+            query_params["ids"] = utils.commajoin(ids)
+
+        return GetScopesResponse(self.get("/v2/api/scopes", query_params=query_params))
+
+    # TODO: Should we relocate this? Would we prefer a different way to pass this
+    # information?
+    class DependentScope(t.NamedTuple):
+        scope: UUIDLike
+        optional: bool
+        requires_refresh_token: bool
```

Comment:
I do think that this needs to exist outside of the class definition; otherwise it will be accessed as a class attribute, like:

```python
dep_scope = AuthClient.DependentScope(...)
```

It may be sufficient to move it to the top of this file and make sure that it's imported all the way up into the `globus_sdk` namespace.

Reply:
I agree -- let's move it out to be separate from the class. Does losing the namespacing from the class require a rename for clarity? `DependentScopeSpec`? (We should discuss the name in any case.)

I think my preference would be to put it into a new module at `src/globus_sdk/services/auth/data.py` as a dedicated class. That's the (informal) convention we have for locating our payload-building helpers.

I would also adjust it to use `globus_sdk.utils.PayloadWrapper` as its base (aside: that class name is _slightly_ amiss at this point, but I don't have a clear plan to rename it yet), which is a dict-like (UserDict). e.g.
```python
from globus_sdk import utils
from globus_sdk._types import UUIDLike

class DependentTokenSpec(utils.PayloadWrapper):
    def __init__(self, *, scope: UUIDLike, optional: bool, requires_refresh_token: bool) -> None:
        super().__init__()
        self["scope"] = str(scope)
        self["optional"] = optional
        self["requires_refresh_token"] = requires_refresh_token
```

The advantage is that with some recent changes in `main` (the same ones which handle `MISSING`), you can then include it in a payload and we'll serialize it before sending. e.g.
```python
self.post(
    "/foo",
    data={"bar": DependentTokenSpec(scope="baz", optional=False, requires_refresh_token=False)}
)
# sends a JSON payload of
# {"bar": {"scope": "baz", "optional": false, "requires_refresh_token": false}}
```